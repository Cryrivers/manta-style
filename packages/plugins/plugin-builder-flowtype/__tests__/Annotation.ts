import getTranspiledString from '../src/utils/transpile';
import { BuilderPluginTestHelper } from '@manta-style/test-helpers';

const testTranspiledString = BuilderPluginTestHelper.testTranspiledString(
  getTranspiledString,
);

describe('Annotation', () => {
  test('Annotation', () => {
    expect(
      testTranspiledString(`type A = {
      /**
       * @mantastyle {{ plugin a b c }}
       */
      a: string;
      /**
       * @mantastyle {{ multiline a
       *  b
       *  c }}
       */
      b: number;
    };`),
    ).toMatchSnapshot();
  });
  test('Handles { and }, escaped bracket', () => {
    expect(
      testTranspiledString(`type A = {
      /**
       * @mantastyle {{ plugin "\\{\\{foo\\}\\}" }}
       */
      a: string;
      /**
       * @mantastyle {{ plugin "\\{\\{foo\\}\\} \\{\\{bar\\}\\}" }}
       */
      b: number;
    };`),
    ).toMatchSnapshot();
  });

  test('Handles }} in other annotation', () => {
    expect(
      testTranspiledString(`type A = {
      /**
       * @mantastyle {{ plugin "\\{\\{foo\\}\\}" }}
       * @random {{}}
       */
      a: string;
      /**
       * @mantastyle {{ plugin "\\{\\{foo\\}\\} \\{\\{bar\\}\\}" }}
       * @random {{}}
       */
      b: number;
    };`),
    ).toMatchSnapshot();
  });
});
