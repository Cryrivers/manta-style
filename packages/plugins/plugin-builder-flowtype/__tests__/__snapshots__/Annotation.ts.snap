// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Annotation Annotation 1`] = `
"FROM:
type A = {
      /**
       * @mantastyle {{ plugin a b c }}
       */
      a: string;
      /**
       * @mantastyle {{ multiline a
       *  b
       *  c }}
       */
      b: number;
    };

--------------------------------------------------------
TO:
'use strict';

var _flowtypeHelpers = _interopRequireDefault(
  require('@manta-style/flowtype-helpers'),
);

var _runtime = _interopRequireDefault(require('@manta-style/runtime'));

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj };
}

var A = _runtime.default.TypeAliasDeclaration(
  'A',
  function(typeFactory) {
    var type = _runtime.default.TypeLiteral(function(typeLiteral) {
      typeLiteral.property(
        'a',
        _runtime.default.StringKeyword,
        false,
        _runtime.default.MantaAnnotation('{{ plugin a b c }}'),
      );
      typeLiteral.property(
        'b',
        _runtime.default.NumberKeyword,
        false,
        _runtime.default.MantaAnnotation('{{ multiline a\\\\nb\\\\nc }}'),
      );
    });

    return type;
  },
  _runtime.default.MantaAnnotation(''),
);
"
`;

exports[`Annotation Handles { and }, escaped bracket 1`] = `
"FROM:
type A = {
      /**
       * @mantastyle {{ plugin \\"{{foo}}\\" }}
       */
      a: string;
      /**
       * @mantastyle {{ plugin \\"{{foo}} {{bar}}\\" }}
       */
      b: number;
    };

--------------------------------------------------------
TO:
'use strict';

var _flowtypeHelpers = _interopRequireDefault(
  require('@manta-style/flowtype-helpers'),
);

var _runtime = _interopRequireDefault(require('@manta-style/runtime'));

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj };
}

var A = _runtime.default.TypeAliasDeclaration(
  'A',
  function(typeFactory) {
    var type = _runtime.default.TypeLiteral(function(typeLiteral) {
      typeLiteral.property(
        'a',
        _runtime.default.StringKeyword,
        false,
        _runtime.default.MantaAnnotation('{{ plugin \\"{{foo}}\\" }}'),
      );
      typeLiteral.property(
        'b',
        _runtime.default.NumberKeyword,
        false,
        _runtime.default.MantaAnnotation('{{ plugin \\"{{foo}} {{bar}}\\" }}'),
      );
    });

    return type;
  },
  _runtime.default.MantaAnnotation(''),
);
"
`;

exports[`Annotation Handles }} in other annotation 1`] = `
"FROM:
type A = {
      /**
       * @mantastyle {{ plugin \\"{{foo}}\\" }}
       * @random {{}}
       */
      a: string;
      /**
       * @mantastyle {{ plugin \\"{{foo}} {{bar}}\\" }}
       * @random {{}}
       */
      b: number;
    };

--------------------------------------------------------
TO:
'use strict';

var _flowtypeHelpers = _interopRequireDefault(
  require('@manta-style/flowtype-helpers'),
);

var _runtime = _interopRequireDefault(require('@manta-style/runtime'));

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj };
}

var A = _runtime.default.TypeAliasDeclaration(
  'A',
  function(typeFactory) {
    var type = _runtime.default.TypeLiteral(function(typeLiteral) {
      typeLiteral.property(
        'a',
        _runtime.default.StringKeyword,
        false,
        _runtime.default.MantaAnnotation('{{ plugin \\"{{foo}}\\" }}'),
      );
      typeLiteral.property(
        'b',
        _runtime.default.NumberKeyword,
        false,
        _runtime.default.MantaAnnotation('{{ plugin \\"{{foo}} {{bar}}\\" }}'),
      );
    });

    return type;
  },
  _runtime.default.MantaAnnotation(''),
);
"
`;
